### PR TITLE
Codacy fix prag

### PR DESCRIPTION
**Pull request by pragamj**
# Problem Statement
- Codacy fixes

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by ketanarlulkar — COMMENTED** [review]
Not all Codacy issues are fixed for libs/prov/provisioner.py & libs/prov/prov_k8s_cortx_deploy.py files. I can see f-strings in loggers.

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 1]
```diff
@@ -463,63 +473,73 @@ def update_sol_yaml(self, worker_obj: list, filepath: str, cortx_image: str,
...
-                                for i in range(0, len(data_devices_f), data_disk_per_cvg)]
-
-            # Create dict for host and disk
-            system_disk = device_list[0]
-            schema = {node_obj.hostname: system_disk}
-            sys_disk_pernode.update(schema)
-        LOGGER.info("Metadata disk %s", metadata_devices)
-        LOGGER.info("data disk %s", data_devices)
+
+        def _cvg_config(data_disk_per_cvg):
```
Comment (seagate-sarang-sawant) [inline libs/prov/prov_k8s_cortx_deploy.py:479]: Same comment as in line 361. Protected can be part of the class itself . Don't you want to reuse it in this module ? Or the functionality is part creating solution yaml itself..

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 2]
```diff
@@ -349,63 +351,71 @@ def deploy_cortx_cluster(self, sol_file_path: str, master_node_list: list,
...
-
-        thread_list = []
-        for each in worker_node_list:
-            t = Thread(target=self.pull_cortx_image, args=(each,))
-            t.start()
-            thread_list.append(t)
-        for each in thread_list:
-            each.join()
-
+        def _operation_on_worker_node():
```
[the remark this answers is not in the document]
Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py:361]: to reduce the local variable

**Review by seagate-sarang-sawant — APPROVED** [review]

**Review by ketanarlulkar — APPROVED** [review]

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 3]
```diff
@@ -1158,12 +1181,97 @@ def check_pods_status(node_obj) -> bool:
...
+            access_key, secret_key = S3H_OBJ.get_local_keys()
+            if self.service_type == "NodePort":
+                s3t_obj = S3TestLib(access_key=access_key, secret_key=secret_key,
+                                    endpoint_url=ext_port_ip)
+            else:
+                s3t_obj = S3TestLib(access_key=access_key, secret_key=secret_key)
+            response = [access_key, secret_key, ext_port_ip]
+        return True, s3t_obj, response
+
+    # pylint: disable=too-many-arguments,too-many-locals,too-many-branches,too-many-statements
```
Comment (sonalk0209) [inline libs/prov/prov_k8s_cortx_deploy.py:1273]: do we want to add pylint disables seperate on each line for better readability purpose?

**Review by sonalk0209 — APPROVED** [review]

**Comment by sonalk0209** [comment]
please add collections output

**Comment by pragamj** [comment]
logs attached to the JIRA https://jts.seagate.com/browse/CORTX-31223

**Pull request merged**